lib/src/util.dart

```diff
@@ -15,4 +18,24 @@ class Util {
         ? truncatedDivision.toString()
         : '$truncatedDivision.$paddingZeros$remainder';
   }
+
+  /// Convert a proto [Date] instance to Dart DateTime.
+  static DateTime convertProtoDateToDateTime(
+          InsertionOrder_Budget_BudgetSegment_DateRange_Date date) =>
+      DateTime(date.year, date.month, date.day);
+
+  /// Convert a string date(yyyy/mm/dd) to Dart DateTime.
+  static DateTime convertStringDateToDateTime(String date) {
+    return DateFormat('yyyy/MM/dd').parse(date);
+  }
+
+  /// Return `true` if [target] is before [start] and after [end],
+  /// false otherwise.
+  static bool isBetweenDates(DateTime target, DateTime start, DateTime end) =>
+      (target.isAfter(start) || target.isAtSameMomentAs(start)) &&
+      (target.isBefore(end) || target.isAtSameMomentAs(end));
+
+  /// Add two string values together.
```

Comment:
Not sure if there is precision loss here. The returned spent from reporting api is a string and the value is not in micros.

Reply:
This is a good callout. consider adding it to the doc comment.

Reply:
How about [decimal](https://pub.dev/packages/decimal), do you think there still might be precision loss there?

Reply:
Decimal should be good!